src/main/java/de/fraunhofer/aisec/cpg/graph/DeclarationSequence.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2019, Fraunhofer AISEC. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+
+package de.fraunhofer.aisec.cpg.graph;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * This represents a sequence of one or more declaration(s). The purpose of this node is primarily
```

Comment:
Declarations can exist without a declaration statement, i.e. directly attached to a translation unit. The problem is that one simple declaration in C++ can actually hold more than one declaration (for example when declaring multiple variables or a class declaration inside a variable declaration).

In the past, we try to work around this by having two distinct handlers, a `DeclarationHandler`, which returns a `Declaration` and a `DeclarationListHandler`, which returns a `List<Declaration>`. This is a constraint, because a handler can either return a type of `N` or  `List<N>` but not both in the same handler.

A C++ simple declaration was first handled by the `DeclarationListHandler`'s handleSimpleDeclaration and then again by the `DeclarationHandler`'s handleSimpleDeclaration, which lead to various errors. My workaround thus was to create a "virtual" CPG node, called a declaration sequence, which is actually a declaration itself and can hold 0..n declarations. Referring to the final graph means that the declaration sequence's children are being added to the parent's list of declarations before translation is finished.